src/charm.py

```diff
@@ -116,6 +116,9 @@ def reconcile(self) -> None:
     @property
     def _pebble_layer(self) -> pebble.LayerDict:
         """Return a dictionary representing a Pebble layer."""
+        # self.state will never be None at this point, but this makes mypy happy.
+        if not self.state:  # pragma: nocover
+            return {}
```

Comment:
Is an empty layer allowed?

Reply:
If this should never be reached, I would suggest to raise an exception instead

Reply:
What do you mean allowed?

Reply:
Why an exception? That's dead code, the issue is that mypy doesn't notice

Reply:
I'm saying that because you can say so in the exceptions message and it seems more semantic to me when it is effectively dead code (and if it goes off for some reason, we don't have bad surprises).

Reply:
What exception would you raise?

Reply:
Something like `Exception("This should never happen")`, I'm not sure if the exception type does matter

Reply:
Turns out the ignore worked, so I'm switching to that instead :)